### PR TITLE
Core: added base64 utility functions

### PR DESCRIPTION
**Pull request by martinfalk**
* https://github.com/ReneNyffenegger/cpp-base64

### CODE REVIEW

File: include/inviwo/core/algorithm/base64.h [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+ *
+ *********************************************************************************/
+#pragma once
+
+#include <inviwo/core/common/inviwocoredefine.h>
+
+#include <string>
+#include <string_view>
+
+namespace inviwo {
```
Comment (ResearchDaniel) [inline include/inviwo/core/algorithm/base64.h]: Might not be necessary to wrap in the inviwo namespace?

**Review by ResearchDaniel — DISMISSED** [review]
LGTM

File: include/inviwo/core/algorithm/base64.h (hunk 1 above)
Comment (petersteneteg) [reply include/inviwo/core/algorithm/base64.h]: I think it should be in the Inviwo namespace, but don't use base64util, just util.
base64util::base64_encode is a bit too much base64. also we usually name it utilbase64.
If you want it that way I would drop the base64 from the function names.

File: include/inviwo/core/algorithm/base64.h [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+// 3. This notice may not be removed or altered from any source distribution.
+
+//
+//  base64 encoding and decoding with C++.
+//  Version: 2.rc.08 (release candidate)
+//
+//  https://github.com/ReneNyffenegger/cpp-base64
+//
+
+IVW_CORE_API std::string base64_encode(std::string const& s, bool url = false);
```
Comment (petersteneteg) [inline include/inviwo/core/algorithm/base64.h]: don't think we need the const string& versions when we have string_view ones...

File: include/inviwo/core/algorithm/base64.h [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+IVW_CORE_API std::string base64_decode(std::string const& s, bool remove_linebreaks = false);
+IVW_CORE_API std::string base64_encode(unsigned char const*, size_t len, bool url = false);
+
+//
+// Interface with std::string_view rather than const std::string&
+// Requires C++17
+// Provided by Yannic Bonenberger (https://github.com/Yannic)
+//
+IVW_CORE_API std::string base64_encode(std::string_view s, bool url = false);
```
Comment (petersteneteg) [inline include/inviwo/core/algorithm/base64.h]: how is the input byte array given? as a string_view? that is a bit odd... span<unsigned char> would have been better probably...

**Review by petersteneteg — CHANGES_REQUESTED** [review]

**Review by petersteneteg — APPROVED** [review]
Nice! like it!

**Pull request merged**